draft-ietf-dnsop-alt-tld.xml

```diff
@@ -125,6 +125,12 @@
       namespace. The .alt label can be used in any domain name as a pseudo-TLD
       to signify that this is an alternative (non-DNS) namespace, and should
       not be looked up in a DNS context.</t>
+      
+      <t>For technical clarity, the on-the-wire format of a name that uses the
```

Comment:
Shouldn't this be "the DNS on-the-wire format"?

Actually, I'm not sure that this sentence really works / helps - the [length]['a','l','t'][null] only occurs within the DNS encoding, and this sentence followed by "Any list of bytes can precede those in the non-DNS name." implies that we are talking about this is the non-DNS context.

E.g: If I'm writing the next big application (TelnetV2!) and a user enters "$> telnet_v2 host.cool.alt", I am likely to do something like:
```
host_name=sys.arg[1]
if name.endswith('cool.alt'):
    ip = cool_resolver.resolve_name(host_name)
...
```

How the 'cool' resolver protocol handles the name in-protocol / on-the-wire doesn't really seem like something that we cover. 

I think that just having the last sentence ("The use of ".alt" in this document...") more clearly covers the intent....

But, I may just be misunderstanding what this is trying to do.

Or, perhaps just earlier "This document reserves that .alt label (0x03616c7400 in DNS wire format) for use as ..."?

Reply:
You are correct that the start should be "the DNS on-the-wire format".

You are incorrect that we can handle user-input to applications in any sane way. We can't even do that now for current DNS names: remember that $.example is a perfectly valid DNS name but would fail in your sample code. That is, because we can't deal with command line input in current DNS, we shouldn't be trying to do that in this document either.

Your proposed wording of `This document reserves that .alt label (0x03616c7400 in DNS wire format) for use as` is not correct because that hex string is two labels. I think that `This document reserves that .alt label followed by the root label (0x03616c7400 in DNS wire format) for use as` is more accurate, but harder to parse than "name... always ends in", but you might disagree.

Reply:
Suggestion:
```
<t>The use of ".alt" in this document uses the presentation format for the DNS,
corresponding to a 0x03616c7400 suffix in DNS wire format.
The presentation and on-the-wire formats for non-DNS protocols might be 
different.</t>
```

Reply:
> We can't even do that now for current DNS names: remember that $.example is a perfectly valid DNS name but would fail in your sample code. That is, because we can't deal with command line input in current DNS, we shouldn't be trying to do that in this document either.

Erm, I wasn't proposing that we discuss command line input -- it was an example to show how I think applications will end up using names -- the `host_name = sys.arg[1]` bit was just shorthand for "some name that the user entered".

Anyway, what do you think of Peter's proposed text. I like it, although I'd probably reformat it slightly as:

```
<t>The use of ".alt" in this document uses the presentation format for the DNS
(corresponding to a 0x03616c7400 suffix in DNS wire format).
The presentation and on-the-wire formats for non-DNS protocols might be 
different.</t>
```
(but then again I'm partial to parenthetical statements...)

Reply:
See latest commit 5b7cf5. I don't like parenthetical statements if they are actually the more clear part. I also added "pseudo-TLD" to make it clear we only mean this in the right-most part of the on-the-wire format. Merge if you like it, let's continue here if you don't.